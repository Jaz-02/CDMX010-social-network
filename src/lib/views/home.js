import {savePost, getPost, signOut, onGetPosts, deletePost, editPost, upDatePost} from "../firebase.js"

export const homeTemplate = async (target) => {
    const html = `
    <div class = "cabeceraHome">  
      <div class = "topHome">
        <div class = "layerHome">
          <img src = "img/encounter-logo.png" alt = "Encounter, the feminist sound space">
        </div>
      </div>
    </div>
    <div class = "main">
      <h1 class="welcome">Bienvenida, comparte tus publicaciones aqu√≠:</h1>
        <div class = "bodyHome">
          <div class = "loginHome"></div>
        </div>
        <form id = "post-form">
          <div>
            <textarea id = "post-title" autofocus></textarea>
          </div>
          <div>
            <textarea id = "post-description"></textarea>
          </div>
          <button id = "btn-post-form">SAVE</button>
        </form>
        <div id = "postConteiner"></div>
    </div>
    <p id= "signOutBtn">SING OFF</p>
    <footer id = "footer-home"><b>encounter</b>, the feminist sound space</footer>
        `;

        target.innerHTML = html
  
        const postForm = document.getElementById("post-form");
        const postContainer = document.getElementById("postConteiner");
        let editStatus = false;
        let id = "";

        onGetPosts((querySnapshot) => {
          const posts = []
          querySnapshot.forEach(doc => {
            // console.log(doc.data())
            const post = doc.data();
            post.id = doc.id;
            posts.push(post);
            
          postContainer.innerHTML = posts.map(post => `
          <div id= "postContainer">
            <div id = "btnContenedor" class="btnContenedor">
              <p id ="btn">...</p>
                <div id= "btnLista" class= "btnLista">
                  <li id="btnEdit" class="btnEdit" data-id="${post.id}" >Edit</li>
                  <li id="btnDelete" class="btnDelete" data-id="${post.id}">Delete</li>              
                </div>
            </div>
            <div class = "postPlace">
              <h3 id="postTitle">${post.title}</h3>
              <p id="postDescription">${post.postDescription}</p>
            </div>
          </div>
          `).join('');

          //DELATE
        const btnDelete = document.querySelectorAll(".btnDelete");
        btnDelete.forEach(btn => {
          btn.addEventListener("click", async (e) => {
           await  deletePost(e.target.dataset.id)
          })
        })

        // EDIT
        const btnsEdit = document.querySelectorAll(".btnEdit");
        btnsEdit.forEach(btn => {
          btn.addEventListener("click", async e => {
            const doc = await editPost(e.target.dataset.id);
            const post = doc.data();
            editStatus = true;
            id = doc.id;
            postForm["post-title"].value = post.title;
            postForm["post-description"].value = post.postDescription;
            postForm["btn-post-form"].innerText ="Update"
          })
        })
      });
    })
        //save
        postForm.addEventListener("submit", async (e) => {
          e.preventDefault();
          const title = postForm["post-title"];
          // console.log(title.value);
          const postDescription = postForm["post-description"];
          // console.log(postDescription.value);
          if (!editStatus) {
          await savePost (title.value, postDescription.value);
          } else {
            await upDatePost (id, {
              title: title.value,
              postDescription: postDescription.value

            })
          }

      //     let likeBtns= document.querySelectorAll(".likeBtn");
      //   let countLike = document.querySelectorAll(".inputLike");
      //   likeBtns.forEach(btn => {
      //   btn.addEventListene("click", () => {
      //     countLike.value = parseInt(countLike.value) + 1;
      //     countLike.style.color = "#12ff00"
      //   });
      // })
       
      // let countClick = 0;  
      // function counClickAdd(){
      //   countClick += 1;
      //  }
      //  $("")

  
          postForm.reset();

          title.focus();
        });

        const signOff = document.getElementById("signOutBtn");
        signOff.addEventListener("click", ()  => {
          signOut()
        });
        };

        // <button id="likeBtn" class="likeBtn">
        //     <i class= "fa fa-thumbs-up"></i>
        //   </button>
        //   <input type="number" id="inputLike" class= "inputLike" value = "0" name = "">
        //   </div>