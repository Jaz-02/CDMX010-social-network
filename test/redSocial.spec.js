// importamos la funcion que vamos a testear
import { homeTemplate } from '../src/lib/views/home.js'

describe('Test post', ()=>{
	beforeEach(() => {
		document.body.innerHTML = '<div id="root"></div>';
	});

	test('deberia renderizar', async () => {
		const target = document.getElementById('root');
		const posts = [{ 
			id: 'dhdshjdhs32736743',
			title: 'test', 
			postDescription: 'Aquí el mensaje de tu post',
			likes: []
		}]; 
		const getPostById = jest.fn().mockImplementation(() => Promise.resolve(posts[0]));
		const getAllPosts = jest.fn().mockImplementation(() => Promise.resolve(posts));
		const getUser = jest.fn().mockImplementation(() => {
			return { email: 'test@laboratoria.la', name: 'test' }
		});
		const firebase = { getPostById, getUser, getAllPosts };
		await homeTemplate (target, firebase);
		expect (target.innerHTML).toMatchSnapshot();
	});

	test('comportamiento del boton', async () => {
		const target = document.getElementById('root');
		const posts = [{ 
			id: 'dhdshjdhs32736743',
			title: 'test', 
			postDescription: 'Aquí el mensaje de tu post',
			likes: []
		}]; 
		const getPostById = jest.fn().mockImplementation(() => Promise.resolve(posts[0]));
		const getAllPosts = jest.fn().mockImplementation(() => Promise.resolve(posts));
		const getUser = jest.fn().mockImplementation(() => {
			return { email: 'test@laboratoria.la', name: 'test' }
		});
		const firebase = { getPostById, getUser, getAllPosts };
		await homeTemplate (target, firebase);
		
		const buttonSave = document.getElementById("btn-post-form");
		buttonSave.click();
		expect (target.innerHTML).toMatchSnapshot();
	});

	test('comportamiento del botton signOut', async () => {
		const target = document.getElementById('root');
		const posts = [{ 
			id: 'dhdshjdhs32736743',
			title: 'test', 
			postDescription: 'Aquí el mensaje de tu post',
			likes: []
		}]; 
		const loginTemplate = [{
			emailSignIn: 'test@laboratoria.la',
			passwordSignIn: '123456789'
		}];
		const getPostById = jest.fn().mockImplementation(() => Promise.resolve(posts[0]));
		const getAllPosts = jest.fn().mockImplementation(() => Promise.resolve(posts));
		const getUser = jest.fn().mockImplementation(() => {
			return { email: 'test@laboratoria.la', name: 'test' }
		});
		const signOut  = jest.fn().mockImplementation(() => Promise.resolve(loginTemplate));
		const firebase = { getPostById, getUser, getAllPosts, signOut };
		await homeTemplate (target, firebase);
	
		const buttonExit = document.getElementById("signOutBtn");
			buttonExit.click();
			expect (target.innerHTML).toMatchSnapshot();
	});

	test('debe renderizar likePost', async () => {
		const target = document.getElementById('root');
		const posts = [{ 
			id: 'dhdshjdhs32736743',
			title: 'test', 
			postDescription: 'Aquí el mensaje de tu post',
			likes: []
		}]; 
		
		const getPostById = jest.fn().mockImplementation(() => Promise.resolve(posts[0]));
		const getAllPosts = jest.fn().mockImplementation(() => Promise.resolve(posts));
		const getUser = jest.fn().mockImplementation(() => {
			return { email: 'test@laboratoria.la', name: 'test' }
		});
		const likePost = jest.fn().mockImplementation(() => {
			return { id: "dflkuutd3456", email: 'test@laboratoria.la' }
		});

		const firebase = { getPostById, getUser, getAllPosts, likePost };
		await homeTemplate (target, firebase);
		expect (target.innerHTML).toMatchSnapshot();
	});

	test('comportamiento del botton like', async () => {
		const target = document.getElementById('root');
		const posts = [{ 
			id: 'dhdshjdhs32736743',
			title: 'test', 
			postDescription: 'Aquí el mensaje de tu post',
			likes: []
		}]; 
		
		const getPostById = jest.fn().mockImplementation(() => Promise.resolve(posts[0]));
		const getAllPosts = jest.fn().mockImplementation(() => Promise.resolve(posts));
		const getUser = jest.fn().mockImplementation(() => {
			return { email: 'test@laboratoria.la', name: 'test' }
		});
		const likePost = jest.fn().mockImplementation(() => {
			return { id: "dflkuutd3456", email: 'test@laboratoria.la' }
		});
		
		const firebase = { getPostById, getUser, getAllPosts, likePost };
		await homeTemplate (target, firebase);
		
		const buttonLike = document.getElementById("likeBtn");
			buttonLike.click();
			expect (target.innerHTML).toMatchSnapshot();
	});



});





