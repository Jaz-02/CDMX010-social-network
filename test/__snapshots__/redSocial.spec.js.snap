// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test post comportamiento del boton 1`] = `
"
  <div class=\\"cabeceraHome\\">  
  <div class=\\"topHome\\">
  <div class=\\"layerHome\\">
  <img src=\\"img/encounter-logo.png\\" alt=\\"Encounter, the feminist sound space\\">
  </div>
  </div>
  </div>
  <div class=\\"main\\">
  <h1 class=\\"welcome\\">Bienvenida, comparte tus publicaciones aquí:</h1>
  <div class=\\"bodyHome\\">
  <div class=\\"loginHome\\"></div>
  </div>
  <form id=\\"post-form\\">
  <div>
  <textarea id=\\"post-title\\" autofocus=\\"\\" required=\\"\\"></textarea>
  </div>
  <div>
  <textarea id=\\"post-description\\" required=\\"\\"></textarea>
  </div>
  <button id=\\"btn-post-form\\">SAVE</button>
  </form>
  <div id=\\"postConteiner\\">
  <div id=\\"postContainer\\">
  <div id=\\"btnContenedor\\" class=\\"btnContenedor\\">
  <p id=\\"btn\\">...</p>
  <div id=\\"btnLista\\" class=\\"btnLista\\">
  <li id=\\"btnEdits\\" class=\\"btnEdit\\" data-id=\\"dhdshjdhs32736743\\">Edit</li>
  <li id=\\"btnDeletes\\" class=\\"btnDelete\\" data-id=\\"dhdshjdhs32736743\\">Delete</li>              
  </div>
  </div>
  <div class=\\"postPlace\\">
  <h3 id=\\"postTitle\\">test</h3>
  <p id=\\"postDescription\\">Aquí el mensaje de tu post</p>
  </div>
  <div class=\\"containerLikeBtn\\">
  <button id=\\"likeBtn\\" class=\\"likeButtons\\" data-id=\\"dhdshjdhs32736743\\">
  <i class=\\"fa fa-thumbs-up\\"></i>
  </button>
  <div class=\\"inputLikes\\" data-id=\\"dhdshjdhs32736743\\">0</div>
  </div>
  </div>
  </div>
  </div>
  <p id=\\"signOutBtn\\">SING OFF</p>
  <footer id=\\"footer-home\\"><b>encounter</b>, the feminist sound space</footer>
  "
`;

exports[`Test post comportamiento del botton like 1`] = `
"
  <div class=\\"cabeceraHome\\">  
  <div class=\\"topHome\\">
  <div class=\\"layerHome\\">
  <img src=\\"img/encounter-logo.png\\" alt=\\"Encounter, the feminist sound space\\">
  </div>
  </div>
  </div>
  <div class=\\"main\\">
  <h1 class=\\"welcome\\">Bienvenida, comparte tus publicaciones aquí:</h1>
  <div class=\\"bodyHome\\">
  <div class=\\"loginHome\\"></div>
  </div>
  <form id=\\"post-form\\">
  <div>
  <textarea id=\\"post-title\\" autofocus=\\"\\" required=\\"\\"></textarea>
  </div>
  <div>
  <textarea id=\\"post-description\\" required=\\"\\"></textarea>
  </div>
  <button id=\\"btn-post-form\\">SAVE</button>
  </form>
  <div id=\\"postConteiner\\">
  <div id=\\"postContainer\\">
  <div id=\\"btnContenedor\\" class=\\"btnContenedor\\">
  <p id=\\"btn\\">...</p>
  <div id=\\"btnLista\\" class=\\"btnLista\\">
  <li id=\\"btnEdits\\" class=\\"btnEdit\\" data-id=\\"dhdshjdhs32736743\\">Edit</li>
  <li id=\\"btnDeletes\\" class=\\"btnDelete\\" data-id=\\"dhdshjdhs32736743\\">Delete</li>              
  </div>
  </div>
  <div class=\\"postPlace\\">
  <h3 id=\\"postTitle\\">test</h3>
  <p id=\\"postDescription\\">Aquí el mensaje de tu post</p>
  </div>
  <div class=\\"containerLikeBtn\\">
  <button id=\\"likeBtn\\" class=\\"likeButtons\\" data-id=\\"dhdshjdhs32736743\\">
  <i class=\\"fa fa-thumbs-up\\"></i>
  </button>
  <div class=\\"inputLikes\\" data-id=\\"dhdshjdhs32736743\\">0</div>
  </div>
  </div>
  </div>
  </div>
  <p id=\\"signOutBtn\\">SING OFF</p>
  <footer id=\\"footer-home\\"><b>encounter</b>, the feminist sound space</footer>
  "
`;

exports[`Test post comportamiento del botton signOut 1`] = `
"
  <div class=\\"cabeceraHome\\">  
  <div class=\\"topHome\\">
  <div class=\\"layerHome\\">
  <img src=\\"img/encounter-logo.png\\" alt=\\"Encounter, the feminist sound space\\">
  </div>
  </div>
  </div>
  <div class=\\"main\\">
  <h1 class=\\"welcome\\">Bienvenida, comparte tus publicaciones aquí:</h1>
  <div class=\\"bodyHome\\">
  <div class=\\"loginHome\\"></div>
  </div>
  <form id=\\"post-form\\">
  <div>
  <textarea id=\\"post-title\\" autofocus=\\"\\" required=\\"\\"></textarea>
  </div>
  <div>
  <textarea id=\\"post-description\\" required=\\"\\"></textarea>
  </div>
  <button id=\\"btn-post-form\\">SAVE</button>
  </form>
  <div id=\\"postConteiner\\">
  <div id=\\"postContainer\\">
  <div id=\\"btnContenedor\\" class=\\"btnContenedor\\">
  <p id=\\"btn\\">...</p>
  <div id=\\"btnLista\\" class=\\"btnLista\\">
  <li id=\\"btnEdits\\" class=\\"btnEdit\\" data-id=\\"dhdshjdhs32736743\\">Edit</li>
  <li id=\\"btnDeletes\\" class=\\"btnDelete\\" data-id=\\"dhdshjdhs32736743\\">Delete</li>              
  </div>
  </div>
  <div class=\\"postPlace\\">
  <h3 id=\\"postTitle\\">test</h3>
  <p id=\\"postDescription\\">Aquí el mensaje de tu post</p>
  </div>
  <div class=\\"containerLikeBtn\\">
  <button id=\\"likeBtn\\" class=\\"likeButtons\\" data-id=\\"dhdshjdhs32736743\\">
  <i class=\\"fa fa-thumbs-up\\"></i>
  </button>
  <div class=\\"inputLikes\\" data-id=\\"dhdshjdhs32736743\\">0</div>
  </div>
  </div>
  </div>
  </div>
  <p id=\\"signOutBtn\\">SING OFF</p>
  <footer id=\\"footer-home\\"><b>encounter</b>, the feminist sound space</footer>
  "
`;

exports[`Test post debe renderizar likePost 1`] = `
"
  <div class=\\"cabeceraHome\\">  
  <div class=\\"topHome\\">
  <div class=\\"layerHome\\">
  <img src=\\"img/encounter-logo.png\\" alt=\\"Encounter, the feminist sound space\\">
  </div>
  </div>
  </div>
  <div class=\\"main\\">
  <h1 class=\\"welcome\\">Bienvenida, comparte tus publicaciones aquí:</h1>
  <div class=\\"bodyHome\\">
  <div class=\\"loginHome\\"></div>
  </div>
  <form id=\\"post-form\\">
  <div>
  <textarea id=\\"post-title\\" autofocus=\\"\\" required=\\"\\"></textarea>
  </div>
  <div>
  <textarea id=\\"post-description\\" required=\\"\\"></textarea>
  </div>
  <button id=\\"btn-post-form\\">SAVE</button>
  </form>
  <div id=\\"postConteiner\\">
  <div id=\\"postContainer\\">
  <div id=\\"btnContenedor\\" class=\\"btnContenedor\\">
  <p id=\\"btn\\">...</p>
  <div id=\\"btnLista\\" class=\\"btnLista\\">
  <li id=\\"btnEdits\\" class=\\"btnEdit\\" data-id=\\"dhdshjdhs32736743\\">Edit</li>
  <li id=\\"btnDeletes\\" class=\\"btnDelete\\" data-id=\\"dhdshjdhs32736743\\">Delete</li>              
  </div>
  </div>
  <div class=\\"postPlace\\">
  <h3 id=\\"postTitle\\">test</h3>
  <p id=\\"postDescription\\">Aquí el mensaje de tu post</p>
  </div>
  <div class=\\"containerLikeBtn\\">
  <button id=\\"likeBtn\\" class=\\"likeButtons\\" data-id=\\"dhdshjdhs32736743\\">
  <i class=\\"fa fa-thumbs-up\\"></i>
  </button>
  <div class=\\"inputLikes\\" data-id=\\"dhdshjdhs32736743\\">0</div>
  </div>
  </div>
  </div>
  </div>
  <p id=\\"signOutBtn\\">SING OFF</p>
  <footer id=\\"footer-home\\"><b>encounter</b>, the feminist sound space</footer>
  "
`;

exports[`Test post deberia renderizar 1`] = `
"
  <div class=\\"cabeceraHome\\">  
  <div class=\\"topHome\\">
  <div class=\\"layerHome\\">
  <img src=\\"img/encounter-logo.png\\" alt=\\"Encounter, the feminist sound space\\">
  </div>
  </div>
  </div>
  <div class=\\"main\\">
  <h1 class=\\"welcome\\">Bienvenida, comparte tus publicaciones aquí:</h1>
  <div class=\\"bodyHome\\">
  <div class=\\"loginHome\\"></div>
  </div>
  <form id=\\"post-form\\">
  <div>
  <textarea id=\\"post-title\\" autofocus=\\"\\" required=\\"\\"></textarea>
  </div>
  <div>
  <textarea id=\\"post-description\\" required=\\"\\"></textarea>
  </div>
  <button id=\\"btn-post-form\\">SAVE</button>
  </form>
  <div id=\\"postConteiner\\">
  <div id=\\"postContainer\\">
  <div id=\\"btnContenedor\\" class=\\"btnContenedor\\">
  <p id=\\"btn\\">...</p>
  <div id=\\"btnLista\\" class=\\"btnLista\\">
  <li id=\\"btnEdits\\" class=\\"btnEdit\\" data-id=\\"dhdshjdhs32736743\\">Edit</li>
  <li id=\\"btnDeletes\\" class=\\"btnDelete\\" data-id=\\"dhdshjdhs32736743\\">Delete</li>              
  </div>
  </div>
  <div class=\\"postPlace\\">
  <h3 id=\\"postTitle\\">test</h3>
  <p id=\\"postDescription\\">Aquí el mensaje de tu post</p>
  </div>
  <div class=\\"containerLikeBtn\\">
  <button id=\\"likeBtn\\" class=\\"likeButtons\\" data-id=\\"dhdshjdhs32736743\\">
  <i class=\\"fa fa-thumbs-up\\"></i>
  </button>
  <div class=\\"inputLikes\\" data-id=\\"dhdshjdhs32736743\\">0</div>
  </div>
  </div>
  </div>
  </div>
  <p id=\\"signOutBtn\\">SING OFF</p>
  <footer id=\\"footer-home\\"><b>encounter</b>, the feminist sound space</footer>
  "
`;
